lib/node_modules/@stdlib/assert/has-same-constructor/README.md

```diff
@@ -0,0 +1,93 @@
+<!--
+
+@license Apache-2.0
+
+Copyright (c) 2018 The Stdlib Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+-->
+
+# hasSameConstructor
+
+> Test if two values have same constructor.
+
+<section class="usage">
+
+## Usage
+
+```javascript
+var isArrayLikeObject = require( '@stdlib/assert/has-same-constructor' );
+```
+
+#### hasSameConstructor( value , value )
+
+Tests if two values have same `constructor`.
+
+<!-- eslint-disable object-curly-newline -->
+
+```javascript
+    > var obj1 = new Date();
+      var obj2 = new Date();
+      console.log(hasSameConstructor(obj1, obj2));
+    true
+
+    > var obj1 = new Date();
+      var obj3 = new String("Hello");
+      console.log(hasSameConstructor(obj1, obj3));
+    false
+
+    > var obj3 = new String("Hello");
+      var obj4 = "World";
+      console.log(hasSameConstructor(obj3, obj4));
+    true
+```
+
+If provided with two values with different `constructors`, the function returns `false`.
+
+</section>
+
+<!-- /.usage -->
+
+<section class="examples">
+
+## Examples
+
+<!-- eslint-disable object-curly-newline, object-curly-spacing, no-empty-function, no-restricted-syntax -->
+
+<!-- eslint no-undef: "error" -->
+
+```javascript
+var hasSameConstructor = require( './../lib' );
+
+var DateObject1 = new Date();
+var DateObject2 = new Date();
+var StringObject = new String("Hello");
```

Comment:
Instead of using the `String` constructor, I suggest use more common examples. E.g., comparing a generic array `[ 1, 2, 3 ]`, with a typed array instance (e.g., `new Int32Array( [ 1, 2, 3 ] )`). You can also use stdlib classes, such as `@stdlib/complex/float64` and `@stdlib/complex/float32`.